Sync/Synchronizers/ClientsSynchronizer.swift

```diff
@@ -158,6 +177,28 @@ public class ClientsSynchronizer: BaseSingleCollectionSynchronizer, Synchronizer
         return deferResult((false, []))
     }
 
+    private func uploadRecords(should: Bool, ifUnmodifiedSince: Timestamp?, toLocalClients localClients: RemoteClientsAndTabs, withServer storageClient: Sync15CollectionClient<ClientPayload>) -> Success {
+        return localClients.getCommands() >>== { clients in
+            for client in clients {
+                storageClient.get(client.guid!) >>== { response in
```

Comment:
If `clients` ever has more than one element, these uploads will proceed in parallel. Which is interesting, isn't it? It should all work…

If you decide not to do so, look at the `walk` utility.